### PR TITLE
Add client format on duration tests

### PR DESCRIPTION
**Pull request by iscai-msft**
# Cadl Ranch Contribution Checklist:

- [x] I have written a [scenario spec](../docs/writing-scenario-spec.md)
- [x] I have **meaningful** `@scenario` names. Someone can look at the list of scenarios and understand what I'm covering.
- [x] I have written a [mock API](../docs/writing-mock-apis.md)
- [x] I have used `@scenarioDoc`s for extra scenario description and to tell people **how to pass** my mock api check.

resolve: https://github.com/Azure/cadl-ranch/issues/288

### CODE REVIEW

**Comment by tadelesh** [comment]
@iscai-msft will we release a new tcgc version to replace the dev version in this PR? or maybe this PR need to wait `@dataFormat`(https://github.com/microsoft/typespec/pull/1796) replace the `@clientFormat`?

**Comment by iscai-msft** [comment]
@tadelesh i think we're good with using dev versions in cadl-ranch, esp in this case bc everyone needs to upgrade to support `@clientFormat` with seconds before the next public release of tsp (to unblock mvad)

**Comment by iscai-msft** [comment]
also @tadelesh can't approve it because i'm the original author, but this PR lgtm, thanks!

**Review by timotheeguerin — APPROVED** [review]

**Comment by iscai-msft** [comment]
@srnagar @m-nash can one of you guys take a look? need a strongly typed language approval. thanks!

File: packages/cadl-ranch-specs/http/azure/client-generator-core/client-format/duration/main.tsp
Comment (m-nash) [inline packages/cadl-ranch-specs/http/azure/client-generator-core/client-format/duration/main.tsp:48]: didn't we agree that we need 2 parameters here?  `units` and `format`.  I think there are two real examples both wanting duration in seconds as the units but one wanting a double and one wanting an int?

Comment (timotheeguerin) [reply packages/cadl-ranch-specs/http/azure/client-generator-core/client-format/duration/main.tsp:48]: or maybe have the format be `"seconds-int"`, `"seconds-float64"`

Comment (m-nash) [reply packages/cadl-ranch-specs/http/azure/client-generator-core/client-format/duration/main.tsp:48]: That could work @timotheeguerin, typically would lean towards 2 params here but the combined single param is less work we can do that.

@iscai-msft I would lean towards tcgc parsing / splitting this into two parameters that get passed to emitters.

Comment (iscai-msft) [reply packages/cadl-ranch-specs/http/azure/client-generator-core/client-format/duration/main.tsp:48]: @m-nash @joheredi is working on that right now, we do have 2 options here: we can wait for him to be done, or we can go ahead and release the test now. the more i think though i think you're right, we should wait until all of the support is into tcgc

File: packages/cadl-ranch-specs/http/azure/client-generator-core/client-format/duration/main.tsp
Comment (srnagar) [inline packages/cadl-ranch-specs/http/azure/client-generator-core/client-format/duration/main.tsp:33]: We'll need multiple routes for different formats - int and double.

File: packages/cadl-ranch-specs/http/azure/client-generator-core/client-format/duration/main.tsp
Comment (srnagar) [inline packages/cadl-ranch-specs/http/azure/client-generator-core/client-format/duration/main.tsp:11]: Just curious why we have _Specs_ as part of the namespace. It's in some tests and not in [others](https://github.com/Azure/cadl-ranch/blob/main/packages/cadl-ranch-specs/http/authentication/api-key/main.tsp#L9).

File: packages/cadl-ranch-specs/http/azure/client-generator-core/client-format/duration/main.tsp
Comment (tadelesh) [reply packages/cadl-ranch-specs/http/azure/client-generator-core/client-format/duration/main.tsp:48]: @iscai-msft after the tcgc update, I could help to update the test then.

File: packages/cadl-ranch-specs/http/azure/client-generator-core/client-format/duration/main.tsp
Comment (tadelesh) [reply packages/cadl-ranch-specs/http/azure/client-generator-core/client-format/duration/main.tsp:33]: Will add the test after tcgc update.

File: packages/cadl-ranch-specs/http/azure/client-generator-core/client-format/duration/main.tsp
Comment (tadelesh) [reply packages/cadl-ranch-specs/http/azure/client-generator-core/client-format/duration/main.tsp:11]: The namespace for cadl ranch test is always corresponding to the test folder structure. For test under `azure`, the folder structure is always the same as the core lib namespace segments, so I guess `_Specs_` is added to prevent any possible collision.

Comment (timotheeguerin) [reply packages/cadl-ranch-specs/http/azure/client-generator-core/client-format/duration/main.tsp:11]: yeah the problem is if we keep it then it conflict with the Azure.Core or other azure library namespace and then having `@scenarioService` will make it think all th eoperations inside of Azure.Core should be scenarios

**Comment by tadelesh** [comment]
Replaced by #316.

**Pull request closed without merging**
